metomi/rose/tests/conftest.py

```diff
@@ -16,15 +13,27 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with Rose. If not, see <http://www.gnu.org/licenses/>.
-# -----------------------------------------------------------------------------
 
-"""This file is used by Travis-CI to start the coverage process.
+from pathlib import Path
+from shutil import rmtree
+from tempfile import TemporaryDirectory
+
+import pytest
+from _pytest.monkeypatch import MonkeyPatch
 
-In order to make python aware of it, we export PYTHONPATH when running
-the tests.
 
-"""
+@pytest.fixture(scope='module')
+def mod_monkeypatch():
+    """A monkeypatch fixture with module-level scope."""
+    patch = MonkeyPatch()
+    yield patch
+    patch.undo()
 
 
-import coverage
-coverage.process_startup()
+@pytest.fixture(scope='module')
+def mod_tmp_path():
+    """A tmp_path fixture with module-level scope."""
+    path = Path(TemporaryDirectory().name)
+    path.mkdir()
+    yield path
+    rmtree(path)
```

Comment:
possible alternative
```suggestion
@pytest.fixture(scope='module')
def mod_tmp_path(tmp_path_factory):
    path = tmp_path_factory.mktemp('foo')
    yield path
```
- Uses pytests inbuilt cleanup
- Rescoped by scope of inheriting fixture - I tried it with tests in different modules and I got test files `basetemp/foo1` and `basetemp/foo2`.